en/developer/reference/articles/02-workflow/05-notification-templates.markdown

```diff
@@ -0,0 +1,287 @@
+| ${expandoTableLocalService}  |          | &#10004;  | [ExpandoTableLocalService](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/expando/kernel/service/ExpandoTableLocalService.html) |
+| ${localeUtil}                |          | &#10004;  | [LocaleUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/LocaleUtil.html) |
+| ${groupId}                   | &#10004; |           | The ID of the group                          |
+| ${portalUtil}                |          | &#10004;  | [PortalUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/PortalUtil.html) |
+| ${validator}                 |          | &#10004;  | [Validator](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/Validator.html) |
+| ${dateUtil}                  |          | &#10004;  | [DateUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/DateUtil.html) |
+| ${serviceLocator}            |          | &#10004;  | [ServiceLocator](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-impl/com/liferay/portal/template/ServiceLocator.html) |
+| ${serviceContext}            |          | &#10004;  | [ServiceContext](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/service/ServiceContext.html) |
+| ${jsonFactoryUtil}           |          | &#10004;  | [JSONFactoryUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/json/JSONFactoryUtil.html) |
+| ${stringUtil}                |          | &#10004;  | [StringUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/StringUtil.html) |
+| ${freeMarkerPortletPreferences} |       | &#10004;  | [TemplatePortletPreferences](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-impl/com/liferay/portal/template/TemplatePortletPreferences.html) |
+| ${dateFormats}              | &#10004;  |           | DESCRIBE |
+| ${dateFormatFactory}         |          | &#10004;  | [DateFormatFactory](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/DateFormatFactory.html) |
+| ${userPortraitURL}           | &#10004; |           | DESCRIBE |
+| ${portal}                    |          | &#10004;  | [Portal](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/Portal.html) |
+| ${commonPermission}          |          | &#10004;  | [CommonPermission](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/service/permission/CommonPermission.html) |
+| ${userURL}                   | &#10004; |           | DESCRIBE |
```

Comment:
@rafaprax another one i could use help with :point_up_2:

Reply:
@rbohl This one is specific for BlogsEntries.  Can we remove this? What do you think?

Reply:
Same as above, I'd say something like

> Applies only to Blogs Entries. It's the URL to the list of Blogs Entries authored by the current User.

Or whatever the actual description is.
@rafaprax

Reply:
Also, how do I determine which assets will have specific variables injected? I might create a separate ticket to add those ones.

Reply:
https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-impl/com/liferay/portal/model/impl/UserImpl.html#getDisplayURL-com.liferay.portal.kernel.theme.ThemeDisplay-

Reply:
Basically, to know the specific variables injected per asset you need to search for every call of `WorkflowHandlerRegistryUtil.startWorkflowInstance` and check what is inserted into the `workflowContext` map variable. For example, 

1. https://github.com/liferay/liferay-portal/blob/master/modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsEntryLocalServiceImpl.java#L2220
2. https://github.com/liferay/liferay-portal/blob/master/modules/apps/blogs/blogs-serhttps://github.com/liferay/liferay-portal/blob/master/modules/apps/wiki/wiki-service/src/main/java/com/liferay/wiki/service/impl/WikiPageLocalServiceImpl.java#L3190

Thx

Reply:
Thanks Rafael. Tracking this [in LRDOCS-7815](https://issues.liferay.com/browse/LRDOCS-7815)